lib/spack/spack/fetch_strategy.py

```diff
@@ -829,9 +829,15 @@ def fetch(self):
             args.append('at revision %s' % self.revision)
         tty.msg("Trying to clone Mercurial repository:", self.url, *args)
 
-        args = ['clone', self.url]
+        args = ['clone']
+
+        if spack.insecure:
+            args.append('--insecure')
```

Comment:
The Package.do_* functions (e.g. do_install()) are the ones meant to be called from commands (vs. install() which is for packagers to override). They also recursively call prior build pipeline stages, so the install command doesn't have to remember to fetch, stage, and patch.  So the place in there where hinge happen is do_stage(), which calls do_fetch and then chdirs into the expanded archive and/or repo directory.  @alalazo surrounded it with a context manager (with pkg.stage) so that the stage is destroyed no matter what happens in test_fetch.  `with` allows you to encapsulate stuff you might put in a try/finally statement.

 The cool thing about that test is that it's using a parameterized fixture.  Since test_fetch asks for the type_of_test fixture, it gets called all the ways that fixture can be parameterized.  So it is called once with type_of_test set to default and once with it set to rev0.  You could add a new test type like insecure and then do the insecure test when it is called that way. Or, you could add another parameter like "secure" and call it with all combinations of  (type_of_test=default, type_of_test=rev0) X (secure=true, secure=False).  

There's docs on that here:

https://docs.pytest.org/en/latest/parametrize.html

Whatever you do, you probably need to validate this by setting spack.insecure before pkg.do_stage and unsetting it after, in a finally block in case anything goes wrong.